### PR TITLE
feat: enabling to generate schema from url and file

### LINKED ISSUE #3: feat: Extract JSON schema from file

### LINKED ISSUE #4: feat: Extract JSON schema from URL

### PR DESCRIPTION
**Pull request by ecojuntak**
close #4 
close #3

### CODE REVIEW

File: goson.go [hunk 1]
```diff
@@ -112,6 +115,24 @@ func GenerateJSONSchema(jsonStr string) (MapString, error) {
...
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	return GenerateJSONSchema(string(body))
+}
+
+func GenerateJSONSchemaFromFile(ir io.Reader) (MapString, error) {
```
Comment (bxcodec) [inline goson.go]: Better to use the param using the path to the file, so we can just pass the path file.

Comment (ecojuntak) [reply goson.go]: I had made like that before, using the file path, but it becomes difficult for me to create the test

Comment (bxcodec) [reply goson.go]: Then add a testfile.json to the repository, and make test against it

Comment (ecojuntak) [reply goson.go]: Okay then, let me fix it

File: goson.go [hunk 2]
```diff
@@ -112,6 +115,25 @@ func GenerateJSONSchema(jsonStr string) (MapString, error) {
...
 }
 
+func GenerateJSONSchemaFromURL(url string) (MapString, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
```
Comment (bxcodec) [inline goson.go]: please don't ignore the error

File: goson.go [hunk 3]
```diff
@@ -112,6 +115,25 @@ func GenerateJSONSchema(jsonStr string) (MapString, error) {
...
+
+	return GenerateJSONSchema(string(body))
+}
+
+func GenerateJSONSchemaFromFile(filePath string) (MapString, error) {
+	file, _ := os.Open(filePath)
```
Comment (bxcodec) [inline goson.go]: please don't ignore the error

File: goson.go [hunk 4]
```diff
@@ -112,6 +115,25 @@ func GenerateJSONSchema(jsonStr string) (MapString, error) {
...
+	body, _ := ioutil.ReadAll(file)
```
Comment (bxcodec) [inline goson.go]: please don't ignore the error

File: goson.go (hunk 3 above)
Comment (bxcodec) [reply goson.go]: and also defer to close the file

Comment (ecojuntak) [reply goson.go]: already fixed

**Review by bxcodec — CHANGES_REQUESTED** [review]
Please fix all the reviews

**Comment by ecojuntak** [comment]
all review already fixed

**Review by bxcodec — APPROVED** [review]
LGTM

**Pull request merged**